CMakeLists.txt

```diff
@@ -141,20 +141,21 @@ set(DCA_LIBS
   symmetrization
   coarsegraining
   ${DCA_CONCURRENCY_LIB}
-  ${DCA_EXTERNAL_LIBS}
   ${DCA_THREADING_LIBS}
 )
 
 if (DCA_HAVE_CUDA)
   list(APPEND DCA_CUDA_LIBS
-    cuda_utils) 
+    cuda_utils)
   list(APPEND DCA_LIBS
     ctaux_walker_kernels
     lapack_kernels
     blas_kernels
     ${DCA_CUDA_LIBS})
 endif()
 
+list(APPEND DCA_LIBS ${DCA_EXTERNAL_LIBS})
```

Comment:
Perhaps add a comment before this line to document the reason for doing this?  Something like:
"# Link external libraries at the end because MAGMA requires the linear algebra libraries in the DCA_EXTERNAL_LIBS list."

Otherwise the reason for doing so will be forgotten overtime.